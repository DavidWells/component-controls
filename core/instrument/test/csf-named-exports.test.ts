import { loadStoriesTests } from './loadTestFiles';

describe('csf-named-exports', () => {
  loadStoriesTests('csf', 'named-exports');
});
