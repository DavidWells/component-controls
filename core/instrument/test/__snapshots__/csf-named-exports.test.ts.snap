// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csf-named-exports export-alias.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T01:03:49.617Z,
    "dateModified": 2020-04-14T04:57:32.955Z,
    "package": "c071cdabd810d210aa3df2ee281eb1e2",
    "title": "Story",
  },
  "packages": Object {
    "c071cdabd810d210aa3df2ee281eb1e2": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.5.1",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "c071cdabd810d210aa3df2ee281eb1e2",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/named-exports/export-alias.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.5.5",
    },
  },
  "stories": Object {
    "exportedStory": Object {
      "arguments": Array [],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 24,
          "line": 2,
        },
        "start": Object {
          "column": 16,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
const myStory = () => {};
myStory.story = {
  name: 'Custom story name',
};

export { myStory as exportedStory };
",
}
`;

exports[`csf-named-exports property-name.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T00:51:47.664Z,
    "dateModified": 2020-04-14T04:57:38.306Z,
    "package": "35c5b6349c3b5c253bb5b98e0b2119fb",
    "title": "Story",
  },
  "packages": Object {
    "35c5b6349c3b5c253bb5b98e0b2119fb": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.5.1",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "35c5b6349c3b5c253bb5b98e0b2119fb",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/named-exports/property-name.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.5.5",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 24,
          "line": 2,
        },
        "start": Object {
          "column": 16,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
const myStory = () => {};
myStory.story = {
  name: 'Custom story name',
};

export { myStory };
",
}
`;

exports[`csf-named-exports re-export-name.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": 2020-03-07T01:05:33.307Z,
    "dateModified": 2020-04-14T04:57:50.819Z,
    "package": "b27d69ca04362a47a347cdfb49a31bf6",
    "title": "Story",
  },
  "packages": Object {
    "b27d69ca04362a47a347cdfb49a31bf6": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.5.1",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "b27d69ca04362a47a347cdfb49a31bf6",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/named-exports/re-export-name.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.5.5",
    },
  },
  "stories": Object {},
  "transformed": "import { myStory } from 'stories.tsx';
export default { title: 'Story' };
myStory.story = {
  name: 'Custom story name',
};

export { myStory };
",
}
`;
